sqlalchemy-stubs/sql/coercions.pyi

```diff
@@ -21,17 +18,19 @@ sqltypes: ModuleType
 traversals: ModuleType
 
 def expect(
-    role: Any,
+    role: _T,
     element: Any,
     apply_propagate_attrs: Optional[Any] = ...,
     argname: Optional[Any] = ...,
     **kw: Any,
-): ...
-def expect_as_key(role: Any, element: Any, **kw: Any): ...
-def expect_col_expression_collection(role: Any, expressions: Any) -> None: ...
+) -> _T: ...
```

Comment:
`T -> T` in general is not correct. Role is a `*Role` class and the returned value depends, but it's not a Role instance.

Reply:
Example: https://github.com/sqlalchemy/sqlalchemy/blob/cf7e345d953cfc276c0008b1b634d459aee19489/lib/sqlalchemy/sql/elements.py#L153
`LiteralValueRole` as role `BindParameter` are return value